packages/contracts-bedrock/src/Safe/LivenessModule.sol

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Safe } from "safe-contracts/Safe.sol";
+import { Enum } from "safe-contracts/common/Enum.sol";
+import { OwnerManager } from "safe-contracts/base/OwnerManager.sol";
+import { LivenessGuard } from "src/Safe/LivenessGuard.sol";
+
+// TODO(maurelian): remove me
+import { console2 as console } from "forge-std/console2.sol";
+
+/// @title LivenessModule
+/// @notice This module is intended to be used in conjunction with the LivenessGuard. It should be able to
+///         execute a transaction on the Safe in only a small number of cases:
+contract LivenessModule {
+    /// @notice The Safe contract instance
+    Safe public safe;
+
+    /// @notice The LivenessGuard contract instance
+    LivenessGuard public livenessGuard;
```

Comment:
Similar to https://github.com/ethereum-optimism/optimism/pull/7540#discussion_r1347563031, I'd suggest including the relationship between safe <> guard <> module in the spec and making these `immutable` or adding setter methods accordingly. In other words, if we change the guard, will we also change the module (and vice versa), or are the two coupled and expected to be changed simultaneously?

There's currently no method to change the guard, so implicitly it seems they are coupled, but we should make this explicit.

Reply:
Agree, I've addressed this by: 
1. making the guard immutable, and 
2. adding a check in `removeGuard` to ensure that the Guard is unchanged.